build.gradle.kts

```
@@ -83,35 +83,36 @@ fun generateTask(taskName: String, incubating: Boolean) {
     standardOutput = System.out
     executable = "docker"
 
-    var filter = if (incubating) "any" else "is_stable"
-    var classPrefix = if (incubating) "Incubating" else ""
+    var target = if (incubating) "incubating_java" else "java"
     val outputDir = if (incubating) "semconv-incubating/src/main/java/io/opentelemetry/semconv/incubating/" else "semconv/src/main/java/io/opentelemetry/semconv/"
-    val packageNameArg = if (incubating) "io.opentelemetry.semconv.incubating" else "io.opentelemetry.semconv"
-    val stablePackageNameArg = if (incubating) "io.opentelemetry.semconv" else ""
 
-    setArgs(listOf(
-        "run",
+    val file_args = if (org.gradle.internal.os.OperatingSystem.current().isWindows())
+      // Don't need to worry about file system permissions in docker.
+      listOf()
+    else {
+      // Make sure we run as local file user
+      val unix = com.sun.security.auth.module.UnixSystem() 
+      val uid = unix.getUid() // $(id -u $USERNAME)
+      val gid = unix.getGid() // $(id -g $USERNAME)
+      listOf("-u", "$uid:$gid")
+    }
+    val weaver_args = listOf(
         "--rm",
```

Apparently the `otel/weaver` image doesn't publish for `arm64` platform yet. 
```suggestion
        "--rm",
        "--platform=linux/x86_64",
```

---

I can fix that, can you open a feature request on weaver?

---

https://github.com/open-telemetry/weaver/issues/345